src/importer.rs

```diff
@@ -334,6 +241,278 @@ impl PyImporter {
+                        } else {
+                            debug!("local asset search path not found");
+                        }
+                    }
+                    _ => {} // Ignore unrecognized parameters for this extraction
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Extract VMF-specific settings with old parameter names
+    pub fn extract_vmf_settings(kwargs: Option<&PyDict>) -> PyResult<VmfSettings> {
+        let mut import_brushes = true;
+        let mut import_overlays = true;
+        let mut epsilon = 0.01;
+        let mut cut_threshold = 0.1;
+        let mut merge_solids = MergeSolids::Merge;
+        let mut invisible_solids = InvisibleSolids::Skip;
+        let mut import_props = true;
+        let mut import_other_entities = true;
+        let mut import_skybox = true;
+        let mut scale = 1.0;
+        let mut unrecognized_keys = Vec::new();
+
+        if let Some(kwargs) = kwargs {
+            for (key, value) in kwargs {
+                if value.is_none() {
+                    continue;
+                }
+
+                let key_str: &str = key.extract()?;
+                match key_str {
+                    "import_brushes" => {
+                        import_brushes = value.extract()?;
+                    }
+                    "import_overlays" => {
+                        import_overlays = value.extract()?;
+                    }
+                    "epsilon" => {
+                        epsilon = value.extract()?;
+                    }
+                    "cut_threshold" => {
+                        cut_threshold = value.extract()?;
+                    }
+                    "merge_solids" => match value.extract()? {
+                        "MERGE" => merge_solids = MergeSolids::Merge,
+                        "SEPARATE" => merge_solids = MergeSolids::Separate,
+                        _ => return Err(PyTypeError::new_err("unexpected kwarg value")),
+                    },
+                    "invisible_solids" => match value.extract()? {
+                        "IMPORT" => invisible_solids = InvisibleSolids::Import,
+                        "SKIP" => invisible_solids = InvisibleSolids::Skip,
+                        _ => return Err(PyTypeError::new_err("unexpected kwarg value")),
+                    },
+                    "import_props" => {
+                        import_props = value.extract()?;
+                    }
+                    "import_entities" => {
+                        import_other_entities = value.extract()?;
+                    }
+                    "import_sky" => {
+                        import_skybox = value.extract()?;
+                    }
+                    "scale" => {
+                        scale = value.extract()?;
+                    }
+                    // Known importer-wide settings, MDL-specific settings, and special filesystem settings - ignore here
+                    "import_materials"
```

Comment:
Could we have separate function for checking if the key is unknown, by having the _ case call that function? This way these don't need to be repeated everywhere.